runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewInternal.java

```diff
@@ -1103,8 +1103,21 @@ private File createImageFile() throws IOException {
         // Create an image file name
         String timeStamp = new SimpleDateFormat("yyyyMMdd_HHmmss").format(new Date());
         String imageFileName = "JPEG_" + timeStamp + "_";
-        File storageDir = Environment.getExternalStoragePublicDirectory(
-                Environment.DIRECTORY_PICTURES);
+        String state = Environment.getExternalStorageState();
+        File storageDir = null;
+
+        // FIXME: If the external storage state is not "MEDIA_MOUNTED", we need to get
+        // other volume paths by "getVolumePaths()" when it was exposed.
+        if (state.equals(Environment.MEDIA_MOUNTED)) {
+            storageDir = Environment.getExternalStoragePublicDirectory(
+                    Environment.DIRECTORY_PICTURES);
+        }
+        if (storageDir == null) {
+            Log.e(TAG, "Can not get the storage directory.");
+            return null;
```

Comment:
If we use the exception, then throw an exception here too, the return value should be always valid.
Or we can catch the IOException inside this method, and return null if the IOException happened, so that the caller doesn't have to deal with the exception.